tiledb/sm/filesystem/azure.cc

```diff
@@ -572,12 +560,25 @@ std::vector<directory_entry> Azure::ls_with_sizes(
   return entries;
 }
 
-void Azure::move_object(const URI& old_uri, const URI& new_uri) {
-  copy_blob(old_uri, new_uri);
-  remove_blob(old_uri);
+void Azure::copy_dir(const URI& old_uri, const URI& new_uri) const {
+  throw_if_not_ok(init_client());
+  auto paths = ls(old_uri, "");
+  while (!paths.empty()) {
+    std::string filename_abs = paths.front();
+    URI filename_uri = URI(filename_abs);
+    std::string filename = filename_abs.substr(old_uri.to_string().length());
+    paths.erase(paths.begin());
+    if (is_dir(filename_uri)) {
+      auto children = ls(filename_uri, "");
+      paths.insert(paths.end(), children.begin(), children.end());
+    } else {
+      URI new_path = URI(new_uri.to_string() + filename);
+      copy_file(filename_uri, new_path);
+    }
+  }
```

Comment:
You are passing an empty delimiter, which means that only files will be returned. This lets you simplify the logic.


```suggestion
  auto paths = ls(old_uri, "");
  while (!paths.empty()) {
    std::string filename_abs = paths.front();
    URI filename_uri = URI(filename_abs);
    std::string filename = filename_abs.substr(old_uri.to_string().length());
    paths.erase(paths.begin());
    URI new_path = URI(new_uri.to_string() + filename);
    copy_file(filename_uri, new_path);
  }
```